### ISSUE TITLE
[Bug]: `Font family 'xkcd' not found` warnings

### ISSUE THREAD

**Issue by maximlt** [matplotlib/matplotlib#26282]
### Bug summary

I created an environment with the latest matplotlib available at the time on conda-forge (3.7.2).

```
conda create -c conda-forge --override-channels -n tempmplfont python=3.9 matplotlib notebook --yes
conda activate tempmplfont
jupyter notebook .
```

I executed the first XKCD example (https://matplotlib.org/stable/gallery/showcase/xkcd.html#sphx-glr-gallery-showcase-xkcd-py). The plot ended up being displayed, however with a bunch of `findfont: Font family 'x' not found.` (cropped output):
![image](https://github.com/matplotlib/matplotlib/assets/35924738/881ed337-ca48-4914-83f9-a184a1491835)

I observed this on my system (Apple M2) and also on Github's CI running on Ubuntu. Apparently I'm not the only one experiencing this issue (https://scikit-rf.readthedocs.io/en/latest/examples/plotting/XKCD%20styling.html):
![image](https://github.com/matplotlib/matplotlib/assets/35924738/4d7a9ca7-ba44-40f2-9dd7-cf84d7a61469)

Also I've just noticed we've been experiencing this issue for a while, except that the warning message seemed to be much shorter and as such easy to miss ([here](https://holoviz.org/tutorial/Plotting.html#save-and-further-customize-plots), this site will be updated soon so rely on the screenshot below please):
![image](https://github.com/matplotlib/matplotlib/assets/35924738/aeb16275-c590-4e50-88e1-fb6d16e6289f)

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np

with plt.xkcd():
    # Based on "Stove Ownership" from XKCD by Randall Munroe
    # https://xkcd.com/418/

    fig = plt.figure()
    ax = fig.add_axes((0.1, 0.2, 0.8, 0.7))
    ax.spines[['top', 'right']].set_visible(False)
    ax.set_xticks([])
    ax.set_yticks([])
    ax.set_ylim([-30, 10])

    data = np.ones(100)
    data[70:] -= np.arange(30)

    ax.annotate(
        'THE DAY I REALIZED\nI COULD COOK BACON\nWHENEVER I WANTED',
        xy=(70, 1), arrowprops=dict(arrowstyle='->'), xytext=(15, -10))

    ax.plot(data)

    ax.set_xlabel('time')
    ax.set_ylabel('my overall health')
    fig.text(
        0.5, 0.05,
        '"Stove Ownership" from xkcd by Randall Munroe',
        ha='center')
```

### Actual outcome

Lots of warnings:
```
findfont: Font family 'xkcd' not found.
findfont: Font family 'xkcd Script' not found.
findfont: Font family 'Humor Sans' not found.
findfont: Font family 'Comic Neue' not found.
findfont: Font family 'xkcd' not found.
findfont: Font family 'xkcd Script' not found.
findfont: Font family 'Humor Sans' not found.
...
```

### Expected outcome

The right font being applied without warning? 

### Matplotlib Version

3.7.2

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.9.16

### Jupyter version

6.5.4

### Installation

conda

**Comment by maximlt** [comment]
Thanks, my mistake indeed!

**Comment by jublicastino** [comment]
> It looks like you are missing the font package - see https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.xkcd.html
> 
> `For best results, the "Humor Sans" font should be installed: it is not included with Matplotlib.`
> 
> Can you try installing it manually, or with your package manager? See also https://matplotlib.org/stable/devel/dependencies.html#additional-external-dependencies. Cheers!

@melissawm 
---
Hello I have same issue with same fonts: i followed the content of 2 links to install manually (in OS windows).
I am using anaconda. I got the following error while running the code in cmd conda Prompt ( iam beginner user so may be I am running the codes in wrong order)
C:\Users\ABC>python -m pip install --config-settings=setup-args="-Dsystem-freetype=true"

****ERROR: You must give at least one requirement to install (see "pip help install")****

o
I have updated the Pip , already installed packages freetype and Qhull...
any idea how to solve the issue with fonts?

additionally, it seems to me I have to change some thing in variable environment in windows?

installation by source file is more complicated and I didnot get what exactly should I do...

**Comment by melissawm** [comment]
The error message is telling you the command you listed above doesn't show a package to install - you are actually missing a dot at the end of the command:

```
python -m pip install --config-settings=setup-args="-Dsystem-freetype=true" .
```

Can you try that? The dot at the end means "install the package from this folder"

**Comment by jublicastino** [comment]
@melissawm 

I tried it. I juts copy paste the code you wrote with period at the end.

python -m pip install --config-settings=setup-args="-Dsystem-freetype=true" .
still there is an error.. see the screenshot.

**ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.**

![cc](https://github.com/user-attachments/assets/67fc18a1-3bd5-4fed-b640-a320e2fbd84e).




just to remind this is the original codes from the link:

python -m pip install \
  --config-settings=setup-args="-Dsystem-freetype=true" \
  --config-settings=setup-args="-Dsystem-qhull=true" \
  .
did I use it in correct way (without backslashes for each configuration?

**Comment by QuLogic** [comment]
Is there a reason you are installing from source? If you want the xkcd fonts to work, all you need to do is install the fonts. The existing wheels are fine to use with that.

**Comment by jublicastino** [comment]
@QuLogic 

fonts have been installed already  but not working .. same error.
i am using anaconda navigator 2.6.1(latest version) with default packages.
should I downgrade my matplotlib? if I know what matplotlib version you use and have such error.

**Comment by QuLogic** [comment]
You should probably delete the font cache at `C:\Users\<username>\.matplotlib\fontlist-*`; there is no need to change versions.

**Comment by jublicastino** [comment]
hello @QuLogic I just followed up to that folder and deleted 2  files.
 I kept the deleted files so i can place it back.

but issue regarding fonts "Comic Neue" solved but another issue popped up this time:

**findfont: Font family 'xkcd' not found.
findfont: Font family 'xkcd Script' not found.
findfont: Font family 'xkcd' not found.
findfont: Font family 'xkcd Script' not found.
findfont: Font family 'xkcd' not found.
findfont: Font family 'xkcd Script' not found**

**Comment by jublicastino** [comment]
just for your information maybe it helps to track the issue
once I deleted 2 files in the that folder: and ran plot codes in jupyter I got a message that matplotlib is building the fonts and it takes time (I cannot remember full message).
after that: i saw a file:
**fontlist-v330.json** placed in that folder.

I opened the file and looked up for comic fonts which was referring to windows folder FONTS.
the comic FONTS as I said I have already installed in the windows.
so I guess probably some thing in JSON file is referring to wrongly?
